File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.swift
class Solution {
    func containsNearbyDuplicate(_ nums: [Int], _ k: Int) -> Bool {
        var window: Set<Int> = []
        var left: Int = 0

        for right in 0..<nums.count {
            if right - left > k {
                window.remove(nums[left])
                left += 1
            }

            if window.contains(nums[right]) {
                return true
            }

            window.insert(nums[right])
        }

        return false
    }
}

File: leetcode/swift/contains-duplicate-ii/0219-contains-duplicate-ii/0219-contains-duplicate-ii.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        recents = {}

        for i, num in enumerate(nums):
            if num in recents and i - recents[num] <= k:
                return True
                
            recents[num] = i

        return False